zaza/openstack/charm_tests/neutron/tests.py

```diff
@@ -352,33 +343,45 @@ def test_get_status_load_balancers(self):
             self.skipTest('LBaasV2 is not supported in this version.')
 
-        # create LBaasV2 for the purpose of this test
-        lbaas_name = 'test_lbaas'
-
-        subnet_list = self.neutron_client.list_subnets(
-            name='private_subnet').get('subnets', [])
-
-        if not subnet_list:
-            raise RuntimeError('Expected subnet "private_subnet" is not '
-                               'configured.')
+        loadbalancer_id = None
 
-        subnet = subnet_list[0]
-        loadbalancer_data = {'loadbalancer': {'name': lbaas_name,
-                                              'vip_subnet_id': subnet['id']}}
-        self.neutron_client.create_loadbalancer(body=loadbalancer_data)
-
-        # test that client and action report same data
-        ngw_unit = zaza.model.get_units(self.application_name,
-                                        model_name=self.model_name)[0]
-        lbaas_from_client = self.neutron_client.list_loadbalancers().get(
-            'loadbalancers', [])
-
-        result = zaza.model.run_action(ngw_unit.entity_id,
-                                       'get-status-lb',
-                                       model_name=self.model_name,
-                                       action_params={"format": "json"})
+        try:
+            # create LBaasV2 for the purpose of this test
+            lbaas_name = 'test_lbaas'
+            subnet_list = self.neutron_client.list_subnets(
+                name='private_subnet').get('subnets', [])
+
+            if not subnet_list:
+                raise RuntimeError('Expected subnet "private_subnet" is not '
+                                   'configured.')
+
+            subnet = subnet_list[0]
+            loadbalancer_data = {'loadbalancer': {'name': lbaas_name,
+                                                  'vip_subnet_id': subnet['id']
+                                                  }
+                                 }
+            loadbalancer = self.neutron_client.create_loadbalancer(
+                body=loadbalancer_data)
+            loadbalancer_id = loadbalancer['loadbalancer']['id']
+
+            # test that client and action report same data
+            ngw_unit = zaza.model.get_units(self.application_name,
+                                            model_name=self.model_name)[0]
+            lbaas_from_client = self.neutron_client.list_loadbalancers().get(
+                'loadbalancers', [])
 
-        self._assert_result_match(result, lbaas_from_client, 'load-balancers')
+            result = zaza.model.run_action(ngw_unit.entity_id,
+                                           'get-status-lb',
+                                           model_name=self.model_name,
+                                           action_params={"format": "json"})
+
+            self._assert_result_match(result, lbaas_from_client,
+                                      'load-balancers')
+        except Exception as exc:
+            raise exc
```

Comment:
This isn't needed; the exception with happen if no exceptions are caught which is essentially the same thing.